### PR TITLE
kind.sh: fedora cleanups

### PR DESCRIPTION
**Pull request by squeed**
Handle a few of the quirks that come up when using podman instead of docker.

Signed-off-by: Casey Callendrello <cdc@redhat.com>

### CODE REVIEW

**Comment by squeed** [comment]
@aojea PTAL. Shouldn't be too ugly.

File: contrib/kind.sh [hunk 1]
```diff
@@ -281,6 +282,23 @@ set_default_params() {
   fi
   OVN_HOST_NETWORK_NAMESPACE=${OVN_HOST_NETWORK_NAMESPACE:-ovn-host-network}
 
+  USING_PODMAN=0
+
+  if docker help | grep -q podman; then
+    echo "docker is aliased to podman"
```
Comment (aojea) [inline contrib/kind.sh:288]: does this work for you?
I think that it always uses docker, if you want to use podman you have to set the environment variable
```
sudo KIND_EXPERIMENTAL_PROVIDER=podman kind create cluster
```

Comment (squeed) [reply contrib/kind.sh:288]: I have the fedora package installed that aliases docker to podman

Comment (squeed) [reply contrib/kind.sh:288]: (and by installing this package, kind uses podman without needing any overrides)

Comment (aojea) [reply contrib/kind.sh:288]: but if you only have podman and you don't have docker installed, kind will use podman

Comment (squeed) [reply contrib/kind.sh:288]: right but the script still requires that `docker` be meaningful.

Comment (aojea) [reply contrib/kind.sh:288]: does it? I think that will be a bug, my memory is not good, but I think that we supported the case where docker is not present and only podman is

Comment (squeed) [reply contrib/kind.sh:288]: It uses `docker build` as well as `docker exec`, so it definitely needs `docker` to work.

Comment (squeed) [reply contrib/kind.sh:288]: Huh. Works fine on my machine :-)

Comment (aojea) [reply contrib/kind.sh:288]: are you really using podman with kind?

Comment (trozet) [reply contrib/kind.sh:288]: i think we should just use KIND_EXPERIMENTAL_PROVIDER=podman and not alias or use variables to mask the kind/docker executable with sudo. Force the user to run as root.

Comment (squeed) [reply contrib/kind.sh:288]: basically, running `make` as root kind of stinks, which is why I initially wrote this.

Comment (aojea) [reply contrib/kind.sh:288]: rootless support is very new in kind, and works both in docker and podman, actually docker was first in supporting it 
https://kind.sigs.k8s.io/docs/user/rootless/

However, rootless multinode wasn't working it uses this SLIRP thing to emulate the network

**Comment by dcbw** [comment]
@squeed rebase?

**Pull request closed without merging**